### PR TITLE
STORM-2760: Add Blobstore Migration Scripts

### PR DESCRIPTION
**Pull request by knusbaum**
Add code and helper scripts for migrating active Storm clusters from a locally-backed BlobStore to an HDFS-backed BlobStore.

### CODE REVIEW

File: external/storm-blobstore-migration/pom.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>log4j-over-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <!-- <dependency> -->
+        <!--     <artifactId>storm-core</artifactId> -->
+        <!--     <groupId>org.apache.storm</groupId> -->
+        <!--     <version>0.10.2.y</version> -->
+        <!-- </dependency> -->
```
Comment (revans2) [inline external/storm-blobstore-migration/pom.xml]: Can we remove the commented out code?

File: external/storm-blobstore-migration/pom.xml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-jar-plugin</artifactId>
+				<configuration>
+					<archive>
+						<manifest>
+							<mainClass>org.apache.storm.blobstore.MigratorMain</mainClass>
+						</manifest>
+					</archive>
+				</configuration>
+			</plugin>
```
Comment (revans2) [inline external/storm-blobstore-migration/pom.xml]: Indentation looks really off here.

File: external/storm-blobstore-migration/src/main/java/org/apache/storm/blobstore/ListHDFS.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package org.apache.storm.blobstore;
```
Comment (revans2) [inline external/storm-blobstore-migration/src/main/java/org/apache/storm/blobstore/ListHDFS.java]: Needs a license header.

File: external/storm-blobstore-migration/migrate.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
Comment (revans2) [inline external/storm-blobstore-migration/migrate.sh]: needs a license header

File: external/storm-blobstore-migration/pom.xml [hunk 5]
```diff
@@ -0,0 +1,126 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (revans2) [inline external/storm-blobstore-migration/pom.xml]: needs a license header (all of the files do)

File: external/storm-blobstore-migration/src/main/java/org/apache/storm/blobstore/ListHDFS.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+import org.apache.storm.Config;
+import org.apache.storm.blobstore.ClientBlobStore;
+import org.apache.storm.hdfs.blobstore.HdfsBlobStore;
+import org.apache.storm.hdfs.blobstore.HdfsClientBlobStore;
+import org.apache.storm.utils.Utils;
+
+public class ListHDFS {
+    
+    public static void main(String[] args) throws Exception {
```
Comment (revans2) [inline external/storm-blobstore-migration/src/main/java/org/apache/storm/blobstore/ListHDFS.java]: Could you file a follow on JIRA to migrate this to be a part of the storm admin command?

File: external/storm-blobstore-migration/src/main/java/org/apache/storm/blobstore/ListHDFS.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        if(args.length >= 3) {
+        	System.out.println("SETTING HDFS KEYTAB!");
+        	hdfsConf.put(Config.BLOBSTORE_HDFS_KEYTAB, args[2]);
+        }
+        
+        /* CREATE THE BLOBSTORES */
+        System.out.println("Creating HDFS blobstore.");
+        HdfsBlobStore hdfsBlobStore = new HdfsBlobStore();
+        hdfsBlobStore.prepare(hdfsConf, null, null);
+        System.out.println("Done.");
```
Comment (revans2) [inline external/storm-blobstore-migration/src/main/java/org/apache/storm/blobstore/ListHDFS.java]: I'm not sure that users by default will expect the 3rd Done to mean we really are done.  Could you make the messages a bit more descriptive? or remove them?

This goes for listing the local blobs as well.

File: external/storm-blobstore-migration/src/main/java/org/apache/storm/blobstore/MigrateBlobs.java [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+            InputStream in = bsFrom.getBlob(key, whoFrom);
+            System.out.println("COPYING BLOB " + key + " FROM " + bsFrom + " TO " + bsTo);
+            bsTo.createBlob(key, in, meta, whoTo);
+            System.out.println("DONE CREATING BLOB " + key);
+        }
+    }
+    
+    
+    public static void main(String[] args) throws Exception {
+        // TODO Auto-generated method stub
```
Comment (revans2) [inline external/storm-blobstore-migration/src/main/java/org/apache/storm/blobstore/MigrateBlobs.java]: Please remove this comment.

File: external/storm-blobstore-migration/src/test/java/org/apache/storm/blobstore/AppTest.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
+package org.apache.storm.blobstore;
+
+import junit.framework.Test;
+import junit.framework.TestCase;
+import junit.framework.TestSuite;
+
+/**
+ * Unit test for simple App.
+ */
+public class AppTest
```
Comment (revans2) [inline external/storm-blobstore-migration/src/test/java/org/apache/storm/blobstore/AppTest.java]: Can we remove this file.  Having a bogus test is worse then no test at all.

File: external/storm-blobstore-migration/src/main/java/org/apache/storm/blobstore/ListHDFS.java (hunk 6 above)
Comment (knusbaum) [reply external/storm-blobstore-migration/src/main/java/org/apache/storm/blobstore/ListHDFS.java]: [STORM-2763](https://issues.apache.org/jira/browse/STORM-2763)

**Comment by knusbaum** [comment]
integration test failure looks unrelated.

**Pull request merged**